mobile/test/common/integration/client_integration_test.cc

```diff
@@ -271,10 +271,6 @@ TEST_P(ClientIntegrationTest, BasicHttp2) {
 
 // Do HTTP/3 without doing the alt-svc-over-HTTP/2 dance.
 TEST_P(ClientIntegrationTest, Http3WithQuicHints) {
-  if (version_ != Network::Address::IpVersion::v4) {
-    // Loopback resolves to a v4 address.
-    return;
-  }
```

Comment:
Does this PR fix this issue?

Reply:
same question as Ryan, this seems unrelated?

Reply:
yep!  the problem was the first instance of the test would store "lyft is 127.0.0.1:1234"
and the second instance of the test would store "lyft is 127.0.0.1:5678" but it wasn't _read_ because we'd latched the singleton.

Reply:
it already did resolve to different objects, but due to the singleton latching it on start-up we only ever used one.

Reply:
yup, makes sense now, thanks!